src/function/table/pragma_detailed_profiling_output.cpp

```diff
@@ -0,0 +1,123 @@
+
+	return make_unique<PragmaDetailedProfilingOutputData>(return_types);
+}
+
+unique_ptr<FunctionOperatorData> PragmaDetailedProfilingOutputInit(ClientContext &context,
+                                                                   const FunctionData *bind_data,
+                                                                   vector<column_t> &column_ids,
+                                                                   TableFilterCollection *filters) {
+	return make_unique<PragmaDetailedProfilingOutputOperatorData>();
+}
+
+static void SetValue(DataChunk &output, int index, int op_id, int fun_id, string name, double time) {
+	output.SetValue(0, index, op_id);
+	output.SetValue(1, index, fun_id);
+	output.SetValue(2, index, move(name));
+	output.SetValue(3, index, time);
+}
+
+static void ExtractExpressions(ChunkCollection &collection, ExpressionInformation &info, DataChunk &chunk, int op_id,
+                               int &fun_id, int sample_tuples_count) {
+	if (info.hasfunction) {
+		SetValue(chunk, chunk.size(), op_id, fun_id++, info.function_name, double(info.time) / sample_tuples_count);
+		chunk.SetCardinality(chunk.size() + 1);
+		if (chunk.size() == STANDARD_VECTOR_SIZE) {
+			collection.Append(chunk);
+			chunk.Reset();
+		}
+	}
+	if (info.children.empty()) {
+		return;
+	}
+	// extract the children of this node
+	for (auto &child : info.children) {
+		ExtractExpressions(collection, *child, chunk, op_id, fun_id, sample_tuples_count);
+	}
+	return;
+}
+
+static void PragmaDetailedProfilingOutputFunction(ClientContext &context, const FunctionData *bind_data_p,
+                                                  FunctionOperatorData *operator_state, DataChunk &output) {
+	auto &state = (PragmaDetailedProfilingOutputOperatorData &)*operator_state;
+	auto &data = (PragmaDetailedProfilingOutputData &)*bind_data_p;
+	if (!state.initialized) {
+		// create a ChunkCollection
+		auto collection = make_unique<ChunkCollection>();
+
+		DataChunk chunk;
+		chunk.Initialize(data.types);
+
+		int operator_counter = 1;
+		//        SetValue(output, total_counter++, 0, 0, "Query: " + context.prev_profiler.query,
+		//                 context.prev_profiler.main_query.Elapsed());
+		if (!context.prev_profilers.empty()) {
+			for (auto op : context.prev_profilers.back().second.GetTreeMap()) {
+				int function_counter = 1;
+				if (op.second->info.has_executor) {
+					for (auto &info : op.second->info.executors_info->roots) {
+						ExtractExpressions(*collection, *info, chunk, operator_counter, function_counter,
+						                   op.second->info.executors_info->sample_tuples_count);
+					}
+				}
+				operator_counter++;
+			}
+		}
+		collection->Append(chunk);
+		data.collection = move(collection);
+		state.initialized = true;
+	}
+
+	if (state.offset >= data.collection->Count()) {
+		output.SetCardinality(0);
+		return;
+	}
+	output.Reference(data.collection->GetChunkForRow(state.offset));
```

Comment:
I recommend using GetChunk(chunk_index), and incrementing chunk_index by 1, of doing row-based counting